src/snowcli/cli/connection.py

```diff
@@ -36,38 +48,89 @@ def add(
         "--connection-name",
         prompt="Name for this connection",
         help="Name of the new connection",
+        show_default=False,
     ),
     account: str = typer.Option(
         None,
         "-a",
         "--accountname",
         "--account",
-        prompt="Snowflake account",
-        help="Name assigned to your Snowflake account.",
+        help="Account name to be used to authenticate with Snowflake.",
+        show_default=False,
     ),
     user: str = typer.Option(
         None,
         "-u",
         "--username",
         "--user",
         prompt="Snowflake username",
+        show_default=False,
         help="Username to connect to Snowflake.",
     ),
     password: str = typer.Option(
-        None,
+        EmptyInput(),
         "-p",
         "--password",
+        click_type=OptionalPrompt(),
         prompt="Snowflake password",
         help="Snowflake password",
         hide_input=True,
     ),
+    database: str = typer.Option(
+        EmptyInput(),
+        "-d",
+        "--database",
+        click_type=OptionalPrompt(),
+        prompt="Database for the connection",
+        help="Database to use on Snowflake.",
+    ),
+    schema: str = typer.Option(
+        EmptyInput(),
+        "-s",
+        "--schema",
+        click_type=OptionalPrompt(),
+        prompt="Connection port",
+        help="Schema in use on Snowflake.",
+    ),
+    host: str = typer.Option(
+        EmptyInput(),
+        "-h",
+        "--host",
+        click_type=OptionalPrompt(),
+        prompt="Connection host",
+        help="The host name the connection attempts to connect to.",
+    ),
+    port: int = typer.Option(
+        EmptyInput(),
+        "-P",
+        "--port",
+        click_type=OptionalPrompt(),
+        prompt="Connection port",
+        help="The port to communicate with on the host.",
+    ),
+    region: str = typer.Option(
+        EmptyInput(),
+        "-r",
+        "--region",
+        click_type=OptionalPrompt(),
+        prompt="Snowflake region",
+        help="Region name if not the default Snowflake Database deployment.",
+    ),
```

Comment:
First time see region, do we need this one? 

What about role, warehouse and protocol?

Reply:
Yup, region is supported by snowflake python connector. Good point about role and wh which are more important. Added!

Reply:
My eyes bleeding looking at parameters and connection_entry order but iiyo